staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/compilation.go

```diff
@@ -49,6 +49,10 @@ const (
 	// RuntimeCELCostBudget is the overall cost budget for runtime CEL validation cost per CustomResource
 	//TODO: pick the RuntimeCELCostBudget
 	RuntimeCELCostBudget = math.MaxInt64
+
+	// checkFrequency configures the number of iterations within a comprehension to evaluate
+	// before checking whether the function evaluation has been interrupted
+	checkFrequency = 100
```

Comment:
The checkFrequency configured that we will check the context Done() channel every `checkFrequency`(100) InterpretableCall invocations to halt CEL evaluation when the context is cancelled. Since CEL evaluates on the order of small microseconds, it shouldn't be expensive. I have added [benchmark tests](https://github.com/kubernetes/kubernetes/blob/0ea0fbc099eb4b6b89a3baefa72dd8058cdc2f3d/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation_test.go#L1826-L1913) as well.